tests/test-extensions/classloading-extension/src/main/java/org/mule/test/classloading/CLKeysResolver.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.test.classloading;
+
+import static org.mule.metadata.api.model.MetadataFormat.JAVA;
+import static org.mule.test.classloading.api.ClassLoadingHelper.addClassLoader;
+
+import org.mule.metadata.api.builder.BaseTypeBuilder;
+import org.mule.metadata.api.model.MetadataType;
+import org.mule.runtime.api.connection.ConnectionException;
+import org.mule.runtime.api.metadata.MetadataContext;
+import org.mule.runtime.api.metadata.MetadataKey;
+import org.mule.runtime.api.metadata.MetadataKeyBuilder;
+import org.mule.runtime.api.metadata.MetadataResolvingException;
+import org.mule.runtime.api.metadata.resolving.OutputTypeResolver;
+import org.mule.runtime.api.metadata.resolving.TypeKeysResolver;
+
+import java.util.Set;
+
+import com.google.common.collect.ImmutableSet;
+
+public class CLKeysResolver implements TypeKeysResolver, OutputTypeResolver {
```

Comment:
If you are going to assert that the operation metadata key resolution and source metadata key resolution are using the correct class loader, I would make them use two different resolvers, and use the addClassLoader method with different values.

Reply:
They both end up calling the same method, so I don't feel like there is a need to distinguish between sources and operations by making 2 separate tests.

Reply:
I think that calling the same method is an implementation detail, we should cover both cases.